pkg/operation/botanist/wanted_secrets.go

```diff
@@ -0,0 +1,718 @@
+				fmt.Sprintf("metrics-server.%s.svc", metav1.NamespaceSystem),
+			},
+			IPAddresses: nil,
+
+			CertType:  secrets.ServerClientCert,
+			SigningCA: certificateAuthorities[v1beta1constants.SecretNameCAMetricsServer],
+		},
+
+		// Secret definition for alertmanager (ingress)
+		&secrets.CertificateSecretConfig{
+			Name: common.AlertManagerTLS,
+
+			CommonName:   "alertmanager",
+			Organization: []string{"gardener.cloud:monitoring:ingress"},
+			DNSNames:     b.ComputeAlertManagerHosts(),
+			IPAddresses:  nil,
+
+			CertType:  secrets.ServerCert,
+			SigningCA: certificateAuthorities[v1beta1constants.SecretNameCACluster],
+			Validity:  &endUserCrtValidity,
+		},
+
+		// Secret definition for grafana (ingress)
+		&secrets.CertificateSecretConfig{
+			Name: common.GrafanaTLS,
+
+			CommonName:   "grafana",
+			Organization: []string{"gardener.cloud:monitoring:ingress"},
+			DNSNames:     b.ComputeGrafanaHosts(),
+			IPAddresses:  nil,
+
+			CertType:  secrets.ServerCert,
+			SigningCA: certificateAuthorities[v1beta1constants.SecretNameCACluster],
+			Validity:  &endUserCrtValidity,
+		},
+
+		// Secret definition for prometheus (ingress)
+		&secrets.CertificateSecretConfig{
+			Name: common.PrometheusTLS,
+
+			CommonName:   "prometheus",
+			Organization: []string{"gardener.cloud:monitoring:ingress"},
+			DNSNames:     b.ComputePrometheusHosts(),
+			IPAddresses:  nil,
+
+			CertType:  secrets.ServerCert,
+			SigningCA: certificateAuthorities[v1beta1constants.SecretNameCACluster],
+			Validity:  &endUserCrtValidity,
+		},
+	}
+
+	if !b.Shoot.KonnectivityTunnelEnabled {
+		secretList = append(secretList,
+			// Secret definition for vpn-shoot (OpenVPN server side)
+			&secrets.CertificateSecretConfig{
+				Name: "vpn-shoot",
+
+				CommonName:   "vpn-shoot",
+				Organization: nil,
+				DNSNames:     []string{},
+				IPAddresses:  []net.IP{},
+
+				CertType:  secrets.ServerCert,
+				SigningCA: certificateAuthorities[v1beta1constants.SecretNameCACluster],
+			},
+
+			// Secret definition for vpn-seed (OpenVPN client side)
+			&secrets.CertificateSecretConfig{
+				Name: "vpn-seed",
+
+				CommonName:   "vpn-seed",
+				Organization: nil,
+				DNSNames:     []string{},
+				IPAddresses:  []net.IP{},
+
+				CertType:  secrets.ClientCert,
+				SigningCA: certificateAuthorities[v1beta1constants.SecretNameCACluster],
+			},
+
+			// Secret definition for vpn-tlsauth (OpenVPN tls authentication secret)
+			&secrets.VPNTLSAuthConfig{
+				Name: "vpn-seed-tlsauth",
+			},
+		)
+	}
+
+	var kubecfgToken *secrets.Token
+	if staticToken != nil {
+		var err error
+		kubecfgToken, err = staticToken.GetTokenForUsername(common.KubecfgUsername)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Secret definition for kubecfg
+	secretList = append(secretList, &secrets.ControlPlaneSecretConfig{
+		CertificateSecretConfig: &secrets.CertificateSecretConfig{
+			Name:      common.KubecfgSecretName,
```

Comment:
Hmm, I get the following entry in the ShootState for the kubecfg secret.
```
  - data:
      certificate: null
      privateKey: null
    name: kubecfg
    type: certificate
```

Is that expected?

Reply:
Yep, the kubecfg secret doesn't have a certificate and private key right now. 
TBH we don't really need to add it to the shootstate right now as all of the secret infodata required for the kubecfg secret is in the staticToken and the APIServerBasicAuth infodata. Initially I let it stay but I think I can improve it since there really is no need to have it in the shootstate now that I think about it.

Reply:
Okay, thanks.